lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,934 @@
+          points [0.5, 7., 35.] mm hr-1.
+          All points contain float(1.) except the
+          zero point [0, 0, 7, 7] which is float(0.)
+          and [1:, 0, ...] which are float(0.)
+       self.vii_cube:
+          With extra coordinate of length(3) "threshold" containing
+          points [0.5, 1., 2.] kg m^-2.
+          forecast_period (on time coord): 0.0 hours (simulates nowcast data)
+          Time and forecast_period dimensions "sqeezed" to be Scalar coords.
+          All points contain float(0.)
+"""
+        self.cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(), fp_point=0.0)
+        self.fg_cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(zero_point_indices=[]))
+        self.ltng_cube = add_forecast_reference_time_and_forecast_period(
+            set_up_cube_with_no_realizations(zero_point_indices=[]),
+            fp_point=0.0)
+        self.precip_cube = (
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3), fp_point=0.0))
+        threshold_coord = self.precip_cube.coord('realization')
+        threshold_coord.points = [0.5, 7.0, 35.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('mm hr-1')
+        self.precip_cube.data[1:, 0, ...] = 0.
+        # iris.util.queeze is applied here to demote the singular coord "time"
+        # to a scalar coord.
+        self.vii_cube = squeeze(
+            add_forecast_reference_time_and_forecast_period(
+                set_up_cube(num_realization_points=3,
+                            zero_point_indices=[]),
+                fp_point=0.0))
+        threshold_coord = self.vii_cube.coord('realization')
+        threshold_coord.points = [0.5, 1.0, 2.0]
+        threshold_coord.rename('threshold')
+        threshold_coord.units = cf_units.Unit('kg m^-2')
+        self.vii_cube.data = np.zeros_like(self.vii_cube.data)
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            None)
+        self.assertIsInstance(result, Cube)
+
+    def test_basic_with_vii(self):
+        """Test that the method returns the expected cube type"""
+        plugin = Plugin()
+        result = plugin._modify_first_guess(self.cube,
+                                            self.fg_cube,
+                                            self.ltng_cube,
+                                            self.precip_cube,
+                                            self.vii_cube)
+        self.assertIsInstance(result, Cube)
```

Comment:
Again, you can probably get rid of `test_basic` without vii as it's repeating coverage.